### PR TITLE
Make 'sg setup' interactive on macOS

### PR DESCRIPTION
**Pull request by mrnugget**
This changes the `sg setup` command to a new, interactive model. For now only on macOS. The next step would be to add Linux support, which should be less work, since the "UI" and the user flow is already there.

**I think this is ready to be merged.** There are still `TODO`s in the code and obviously things can always be improved, but I think it's much better than the current `sg setup` we have. Especially because it makes it much clearer what has to happen inside the `sourcegraph` repository and what doesn't.

**I did a complete walkthrough of a setup on a fresh macOS instance and all the steps worked.** The only thing that failed was downloading the `symbols` container. But I'm investigating.

As for the TODOs: I'm going to keep chipping away at them. Please feel free to also pick up some of them. The stuff I want to tackle after this PR:

* Make the "only for employees" clear in the UI
* Linux support
* Better readability of instructions

## Preview

https://user-images.githubusercontent.com/1185253/143875845-a7fc7f73-4ce4-482f-b5ed-47e0a5a53c09.mp4

### CODE REVIEW

**Review by jhchabran — APPROVED** [review]

File: dev/sg/sg_setup.go [hunk 1]
```diff
@@ -426,3 +619,528 @@ func getBool() bool {
...
+		var num int
+		_, err := fmt.Scan(&num)
+		if err != nil {
+			return 0, err
+		}
+
+		if _, ok := idx[num]; ok {
+			return num - 1, nil
+		}
+		fmt.Printf("%d is an invalid choice :( Let's try again?\n", num)
```
Comment (jplahn) [inline dev/sg/sg_setup.go:1020]: I know this is already merged, but I think this is causing the counter [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@13f98ef3eca9513f88eeac38b6476dcbc3bf97c4/-/blob/dev/sg/sg_setup.go?L174:15) to cause the printed command [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@13f98ef3eca9513f88eeac38b6476dcbc3bf97c4/-/blob/dev/sg/sg_setup.go?L185:9) to be off by one. 

Attaching a screen shot where I type in 3 but it decrements by 1 to 2.
image

Really not a big deal usability wise, but noticed it while finally setting up my workspace

Comment (mrnugget) [reply dev/sg/sg_setup.go:1020]: Ugh. Off-by-one errors, my kryptonite. I'll take a look!

Comment (mrnugget) [reply dev/sg/sg_setup.go:1020]: Fix is here: https://github.com/sourcegraph/sourcegraph/pull/29109 Problem was only that it printed the wrong header. "Clone repositories" is 3, so you ended up in the right section, but the header says "2"

Comment (jplahn) [reply dev/sg/sg_setup.go:1020]: 👍

**Pull request merged**